File: ExponeaSDK/Lib/Models/Configuration.cs
using System;
using System.Collections.Generic;

namespace Exponea
{
    public class Configuration
    {

        public Configuration(string projectToken, string authorization, string baseUrl)
        {
            ProjectToken = projectToken;
            Authorization = authorization;
            BaseUrl = baseUrl;
        }

        public string ProjectToken { get; set; }
        public string Authorization { get; set; }
        public string BaseUrl { get; set; }
        public IDictionary<EventType, IList<Project>>? ProjectRouteMap { get; set; }
        public int? MaxTries { get; set; }
        public double? SessionTimeout { get; set; }
        public bool? AutomaticSessionTracking { get; set; }
        public Dictionary<string, Object>? DefaultProperties { get; set; }
        public TokenTrackFrequency TokenTrackFrequency { get; set; } = TokenTrackFrequency.OnTokenChange;
        public HttpLoggingLevel? HttpLoggingLevel { get; set; }
        // ReSharper disable once InconsistentNaming
        public double? CampaignTTL { get; set; }
        public bool AllowDefaultCustomerProperties { get; set; } = true;
        public bool AdvancedAuthEnabled { get; set; } = false;
        public bool AutomaticPushNotification { get; set; } = true;
        public string? PushIcon { get; set; }
        public int? PushAccentColor { get; set; }
        public string? PushChannelName { get; set; }
        public string? PushChannelDescription { get; set; }
        public string? PushChannelId { get; set; }
        public int? PushNotificationImportance { get; set; }
        // ReSharper disable once InconsistentNaming
        public bool? RequirePushAuthorization { get; set; }
        public string? AppGroup { get; set; }
    }
}

File: BloomreachSDKMauiIOS/BloomreachSDKMauiIOS/Sources/Model/Configuration.swift
//
//  Configuration.swift
//  BloomreachSDKMauiIOS
//
//  Created by Ankmara on 28.08.2023.
//

import ExponeaSDK

struct Configuration {

    let appGroup: String
    let advancedAuthEnabled: Bool
    let allowDefaultCustomerProperties: Bool
    let automaticPushNotificationTracking: Bool
    let automaticSessionTracking: Bool
    let baseUrl: String
    let flushEventMaxRetries: Int
    let projectToken: String
    let requirePushAuthorization: Bool
    let sessionTimeout: Double
    let token: String
    var tokenTrackFrequency: TokenTrackFrequency = .onTokenChange

    init(data: [String: Any]) {
        appGroup = .assertValueFromDict(data: data, key: "appGroup")
        advancedAuthEnabled = .assertValueFromDict(data: data, key: "advancedAuthEnabled")
        allowDefaultCustomerProperties = .assertValueFromDict(data: data, key: "allowDefaultCustomerProperties")
        automaticPushNotificationTracking = .assertValueFromDict(data: data, key: "automaticPushNotification")
        automaticSessionTracking = .assertValueFromDict(data: data, key: "automaticSessionTracking")
        baseUrl = .assertValueFromDict(data: data, key: "baseUrl")
        flushEventMaxRetries = data["maxTries"] as? Int ?? 0
        projectToken = .assertValueFromDict(data: data, key: "projectToken")
        requirePushAuthorization = .assertValueFromDict(data: data, key: "requirePushAuthorization")
        sessionTimeout = .assertValueFromDict(data: data, key: "sessionTimeout")
        token = .assertValueFromDict(data: data, key: "authorization")
        if let tokenTrackFrequency = data["tokenTrackFrequency"] as? String {
            switch tokenTrackFrequency {
            case "token_change":
                self.tokenTrackFrequency = .onTokenChange
            case "daily":
                self.tokenTrackFrequency = .daily
            case "every_launch":
                self.tokenTrackFrequency = .everyLaunch
            default:
                assertionFailure("No token track frequency set")
            }
        }
    }
}
